src/kixi/heimdall/handler.clj

```diff
@@ -4,12 +4,63 @@
             [ring.middleware.defaults :refer [wrap-defaults api-defaults]]
             [kixi.heimdall.application :as app]
             [taoensso.timbre :as log]
-            [kixi.heimdall.components.database :as db]            ))
+            [kixi.heimdall.components.database :as db]
+            [buddy.sign.jws :as jws]
+            [buddy.sign.util :as sign]
+            [buddy.core.keys :as ks]
+            [clj-time.core :as t]
+            [clojure.java.io :as io]
+            [kixi.heimdall.user :as user]
+            [clojure.edn :as edn]
+            [environ.core :refer [env]]))
+
+(defn get-config
+  [f]
+  (edn/read-string (slurp f)))
+
+(def auth-config
+  (let [f (io/file (System/getProperty "user.home")
+                   ".heimdall.auth-conf.edn")]
+    (try (get-config f)
+         (catch java.io.FileNotFoundException e
+           (if-let [env-file (:auth-conf env)]
+             (get-config (io/resource env-file))
+             (throw e))))))
+
+(defn- pkey [auth-conf]
+  (ks/private-key
+   (io/resource (:privkey auth-conf))
+   (:passphrase auth-conf)))
+
+(defn create-auth-token [session auth-conf credentials]
+  (let [[ok? res] (user/auth session credentials)
+        exp (-> (t/plus (t/now) (t/days 1)) (sign/to-timestamp))]
+    (if ok?
+      [true {:token (jws/sign (str res)
+                              (pkey auth-conf)
+                              {:alg :rs256 :exp exp})}]
+      [false res])))
+
+(defn auth-token [req]
+  (let [[ok? res] (create-auth-token (:cassandra-session req)
+                                     (:auth-conf req)
+                                     (:params req))]
+    (if ok?
+      {:status 201 :body res}
+      {:status 401 :body res})))
+
+
+(defn wrap-datasource [handler]
+  (fn [req]
+    (handler (assoc req :cassandra-session (:cassandra-session (db/session))))))
```

Comment:
This will create a new session, every time, is that what you want? Also, I've not seen `component/using` used outside the context of a system before - are you sure this is doing what you think? (https://github.com/MastodonC/kixi.heimdall/pull/4/files#diff-c31bf1ede8b51958fbac262f9e4c4a25R112)